cookbooks/bcpc-hadoop/libraries/utils.rb

```diff
@@ -203,59 +203,94 @@ def set_hosts
     'hbase_head' => 'role[BCPC-Hadoop-Head-HBase]'
   }
 
+  srvc2recipe = {
+    'oozie_server' => 'recipe[bcpc-hadoop::oozie]'
+  }
+
   if node.run_state['cluster_def'].nil? then
     node.run_state['cluster_def'] = BACH::ClusterDef.new(node_obj: node)
   end
-  cd = node.run_state['cluster_def']
+  nodes = node.run_state['cluster_def'].fetch_cluster_def
+
+  # host search helper lambdas
+  runs_role = Proc.new { |host, role| host[:runlist].include? srvc2role[role] }
+  runs_recipe = Proc.new { |host, recipe| host[:runlist].include? srvc2recipe[recipe] }
+
+  # mapped host objects
+  to_host = Proc.new do |host| {
```

Comment:
why the need for a similar hash with less keys?

Reply:
I wondered the same thing. I was trying to maintain the original functionality of the code, but utilizing a single hash doesn't hurt here.